apps/escuela-frontend/src/pages/tutoriales/[module]/[exercise]/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+
+import {GetStaticPaths, GetStaticProps} from 'next'
+import {getAllTutorials, getTutorial} from 'lib/tutorials'
+import {getExercise} from 'lib/exercises'
+import {getSection} from 'lib/sections'
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const {params} = context
+  const exerciseSlug = params?.exercise as string
+  const sectionSlug = params?.section as string
+
+  const module = await getTutorial(params?.module as string)
+  const exercise = await getExercise(exerciseSlug)
+  const section = await getSection(sectionSlug)
+
+  return {
+    props: {exercise, module, section},
+    revalidate: 10,
+  }
+}
+
+export const getStaticPaths: GetStaticPaths = async (context) => {
+  const tutorials = await getAllTutorials()
+
+  // flatMap to extract exercises in sections from workshops
+  const paths = tutorials.flatMap((tutorial: any) => {
+    return (
+      tutorial.sections?.flatMap((section: any) => {
+        return (
+          section.exercises?.map((exercise: any) => ({
+            params: {
+              module: tutorial.slug.current,
+              section: section.slug,
+              exercise: exercise.slug,
+            },
+          })) || []
+        )
+      }) || []
+    )
+  })
```

Comment:
@joelhooks how should routes be defined when an exercise contains 1 or more solutions (extra credits)? 

![CleanShot 2022-11-09 at 12 43 14@2x](https://user-images.githubusercontent.com/57044804/200937799-981d29ab-c00a-4f39-bfbe-f07b34175d36.png)